### PR TITLE
chore: automate tagging

### PR DESCRIPTION
**Pull request by gotbadger**
## Description
Tags for release can now be generated automatically by manually triggering the release action.

### How it works
- When the release job is run if no tag ref is given it will generate a new symantic tag. 
- If a tag is created manually the action will continue to work as before.
- Tags can be created from any branch provided that branch is selected when manually triggering the action.

## Checklist

- [ ] I've added test coverage that shows my fix or feature works as expected.
- [ ] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.

### CODE REVIEW

**Comment by gotbadger** [comment]
Need to check a few things before merging this so dont merge for me :)

**Comment by cfabianski** [comment]
Let me know once it's ready please (looks good so far)

**Comment by gotbadger** [comment]
Good to go on this one now

**Review by vjerci — APPROVED** [review]

**Pull request merged**
